dbt/include/athena/macros/materializations/models/table/create_table_as.sql

```diff
@@ -87,3 +87,39 @@
   as
     {{ sql }}
 {% endmacro %}
+
+{% macro create_table_as_with_partitions(temporary, relation, sql) -%}
+
+    {% set partitions_batches = get_partition_batches(sql) %}
+    {% do log('BATCHES TO PROCESS: ' ~ partitions_batches | length) %}
+
+    {%- do log('CREATE EMPTY TABLE: ' ~ relation) -%}
+    {%- set create_empty_table_query -%}
+        {{ create_table_as(temporary, relation, sql) }}
+        limit 0
+    {%- endset -%}
+    {%- do run_query(create_empty_table_query) -%}
+
+    {%- for batch in partitions_batches -%}
+        {%- do log('BATCH PROCESSING: ' ~ loop.index ~ ' OF ' ~ partitions_batches | length) -%}
+
+        {%- set insert_batch_partitions -%}
+            insert into {{ relation }}
+            select * from ({{ sql }})
```

Comment:
I'm not a big fan of select *, do you think that is possible to find a way to get the columns that needed to be used? - in the step above you create an empty table, maybe columns can be picked from there?

Reply:
don't you think this will be an overkill to get column names once again ?

Reply:
If we get them immediately after `{%- do run_query(create_empty_table_query) -%}` and we use them for each insert query is fine - I believe that insert into statements with explicit select and list of columns are making debugging easier - so the overkill is more then welcome.

Reply:
yeah, sure. I've added columns parsing step